### PR TITLE
Delete all versions of DICOM

### PR DESCRIPTION
**Pull request by bcarthic**
## Description
Delete all versions of DICOM.

## Related issues
Addresses [[AB#102006](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/102006)].

## Testing
Added new unit tests and E2E tests

### CODE REVIEW

File: src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/DeleteInstanceV35.sql [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+--     @createdStatus
+--         * Status value representing the created state.
+--     @studyInstanceUid
+--         * The study instance UID.
+--     @seriesInstanceUid
+--         * The series instance UID.
+--     @sopInstanceUid
+--         * The SOP instance UID.
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.DeleteInstanceV35
```
Comment (smithago) [inline src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/DeleteInstanceV35.sql:25]: I don't see any signature change from existing, why is the new version necessary?

File: src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/RetrieveDeletedInstanceV35.sql [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+-- DESCRIPTION
+--     Retrieves deleted instances where the cleanupAfter is less than the current date in and the retry count hasn't been exceeded
+--
+-- PARAMETERS
+--     @count
+--         * The number of entries to return
+--     @maxRetries
+--         * The maximum number of times to retry a cleanup
+/***************************************************************************************/
+CREATE OR ALTER PROCEDURE dbo.RetrieveDeletedInstanceV35
```
Comment (smithago) [inline src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/RetrieveDeletedInstanceV35.sql:17]: same here

File: src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/DeleteInstanceV35.sql (hunk 1 above)
Comment (bcarthic) [reply src/Microsoft.Health.Dicom.SqlServer/Features/Schema/Sql/Sprocs/DeleteInstanceV35.sql:25]: Yeah not needed. Will update it. Just went with a flow

File: src/Microsoft.Health.Dicom.Core/Features/Delete/DeleteService.cs [hunk 3]
```diff
@@ -92,34 +92,43 @@ public async Task<(bool Success, int RetrievedInstanceCount)> CleanupDeletedInst
             try
             {
                 var deletedInstanceIdentifiers = (await _indexDataStore
-                    .RetrieveDeletedInstancesAsync(
+                    .RetrieveDeletedInstancesWithPropertiesAsync(
```
Comment (smithago) [inline src/Microsoft.Health.Dicom.Core/Features/Delete/DeleteService.cs:95]: Won't this fail when code is updated and schema is not updated yet?

In SqlIndexDataStoreV1.cs you will have to call the old procedure and populate the OriginalWatermark as null to make this work seemlessly

Comment (bcarthic) [reply src/Microsoft.Health.Dicom.Core/Features/Delete/DeleteService.cs:95]: I thought I called it, similar to what you did instanceWithProperties. Totally missed it, thanks for catching it

Comment (bcarthic) [reply src/Microsoft.Health.Dicom.Core/Features/Delete/DeleteService.cs:95]: Done

**Review by smithago — APPROVED** [review]

**Pull request merged**